django-backend/fecfiler/scha_transactions/models.py

```diff
@@ -69,3 +79,39 @@ class SchATransaction(SoftDeleteModel, CommitteeOwnedModel):
 
     class Meta:
         db_table = "scha_transactions"
+        indexes = [
+            models.Index(fields=['transaction_id'])
+        ]
+
+    # This is intended to be useable without instantiating a transaction object
+    @staticmethod
+    def check_for_uid_conflicts(uid): # noqa
+        return len(SchATransaction.objects.filter(transaction_id=uid)) > 0
+
+    def generate_unique_transaction_id(self):
+        u = uuid.uuid4()
+        hex_id = u.hex.upper()
+        # Take 20 characters from the end, skipping over the 20th char from the right,
+        # which is the version number (uuid4 -> "4")
+        unique_id = hex_id[-21] + hex_id[-19:]
+
+        attempts = 0
+        while SchATransaction.check_for_uid_conflicts(unique_id):
+            unique_id = str(u.hex).upper()[-20:]
+            attempts += 1
+            if (attempts > 5):
+                logger.info("Unique ID generation failed: Over 5 conflicts in a row")
+                return
+        self.transaction_id = unique_id
+
+    def update_parent_related_values(self, parent):
+        self.back_reference_tran_id_number = parent.transaction_id
```

Comment:
it might be nice to, at some point take these out of the model and populate them in the queryset.  like `back_reference_tran_id_number=parent_transaction__transaction_id`

Reply:
That would definitely be nice to do at some point